services/osparc-gateway-server/src/osparc_gateway_server/backend/osparc.py

```diff
@@ -117,7 +130,10 @@ async def do_check_clusters(self, clusters: List[Cluster]):
     async def do_start_worker(
         self, worker: Worker
     ) -> AsyncGenerator[Dict[str, Any], None]:
-        self.log.debug("received call to start worker as %s", f"{worker=}")
+        self.log.debug("--> starting worker %s", f"{worker=}")
+        node_hostname = await get_next_empty_node_hostname(
```

Comment:
this function raises NoServiceTasksError, ... if e.g. worker is not ready in a node. where do you handle these exceptions? Should you use a retry after some wait-time e.g. in case the worker happen to be starting ...

Reply:
good catch